File: math-easy/944.删列造序.py
#
# @lc app=leetcode.cn id=944 lang=python3
#
# [944] 删列造序
#
from typing import List


# @lc code=start
class Solution:
    def minDeletionSize(self, A: List[str]) -> int:
        ret = 0
        for i in range(len(A[0])):
            for j in range(1, len(A)):
                if (ord(A[j][i]) < ord(A[j-1][i])):
                    ret += 1
                    break
        return ret
# @lc code=end

File: math-easy/944.删列造序.cpp
/*
 * @lc app=leetcode.cn id=944 lang=cpp
 *
 * [944] 删列造序
 */
#include <bits/stdc++.h>
using namespace std;
// @lc code=start
class Solution {
public:
    int minDeletionSize(vector<string> &A) {
        int ret = 0;
        for (int i = 0; i < A[0].length(); i++) {
            for (int j = 0; j < A.size() - 1; j++) {
                if (A[j][i] > A[j + 1][i]) {
                    ret++;
                    break;
                }
            }
        }
        return ret;
    }
};
// @lc code=end
